[[ feature-umaskosx ]] Enable "umask" property under OSX

peter-b opened the pull request:
OS X is basically a BSD and has a POSIX userspace, including umask(2).

Enable setting and retrieving the umask when running on OS X.

peter-b committed:
[[ bugfix-13642 ]] Improve the documentation for the umask property.

When setting (or retrieving) the umask, it is convenient to use octal
representation, and this is what the documentation currently
recommends.

Unfortunately, LiveCode won't convert base-8 literal integers by
default, so at the moment following the instructions in the
documentation and writing "set the umask to 077" gives a real umask of
0115 (octal).

Update the documentation to clarify this a bit.

peter-b committed:
[[ feature-umaskosx ]] Enable "umask" property under OSX

OS X is basically a BSD and has a POSIX userspace, including umask(2).

Enable setting and retrieving the umask when running on OS X.

runrevmark commented:
Hi Peter - did you figure out how to modify the docs yesterday? This will require a small tweak to the umask doc entry to mention its supported on Mac. Thanks!

peter-b commented:
I did figure it out -- see runrev/livecode#1260. What's the best way of dealing with this?  Should I merge that commit into this pull request, then add the mention of OS X?

runrevmark commented:
:) I noticed that pull request after reading this one... Might as well merge that commit into this one and tweak the platforms setting.

peter-b committed:
Merge branch 'bugfix-13642' into feature-umaskosx

peter-b committed:
[[ feature-umaskosx ]] Update "umask" docs to indicate OS X support.

Add the <mac/> platform tag to the umask documentation, but add a note
indicating that it isn't available on Mac OS Classic.

peter-b commented:
Okay, did that. :+1:
